src/Api/Tools/Models/Response/MemberAccessReportModel.cs

```diff
@@ -0,0 +1,150 @@
+/// </summary>
+public class MemberAccessReportAccessDetails
+{
+    public Guid CollectionId { get; set; }
+    public Guid? GroupId { get; set; }
+    public string GroupName { get; set; }
+    public string CollectionName { get; set; }
+    public int ItemCount { get; set; }
+    public bool ReadOnly { get; set; }
+    public bool HidePasswords { get; set; }
+    public bool Manage { get; set; }
+
+    // internal to not expose 
+    internal Guid? UserGuid { get; set; }
+}
+
+/// <summary>
+/// Contains the collections and group collections a user has access to including
+/// the permission level for the collection and group collection. 
+/// </summary>
+public class MemberAccessReportResponseModel
+{
+    public string UserName { get; set; }
+    public string Email { get; set; }
+    public bool TwoFactorEnabled { get; set; }
+    public bool AccountRecoveryEnabled { get; set; }
+    public int GroupsCount { get; set; }
+    public int CollectionsCount { get; set; }
+    public int TotalItemCount { get; set; }
+    public IEnumerable<MemberAccessReportAccessDetails> AccessDetails { get; set; }
+
+    /// <summary>
+    /// Generates a report for all members of an organization. Containing summary information
+    /// such as item, collection, and group counts. As well as detailed information on the
+    /// user and group collections along with their permissions
+    /// </summary>
+    /// <param name="orgUsers">Organization user details collection</param>
+    /// <param name="orgGroups">Organization groups collection</param>
+    /// <param name="orgCollectionsWithAccess">Collections for the organization and the groups/users and permissions</param>
+    /// <param name="orgItems">Cipher items for the organization with the collections associated with them</param>
+    /// <param name="organizationUsersTwoFactorEnabled">Organization users two factor status</param>
+    /// <param name="orgAbility">Organization ability for account recovery status</param>
+    /// <returns>List of the MemberAccessReportResponseModel</returns>;
+    public static IEnumerable<MemberAccessReportResponseModel> CreateReport(
+        IEnumerable<OrganizationUserUserDetails> orgUsers,
+        ICollection<Group> orgGroups,
+        ICollection<Tuple<Collection, CollectionAccessDetails>> orgCollectionsWithAccess,
+        IEnumerable<CipherOrganizationDetailsWithCollections> orgItems,
+        IEnumerable<(OrganizationUserUserDetails user, bool twoFactorIsEnabled)> organizationUsersTwoFactorEnabled,
+        OrganizationAbility orgAbility)
+    {
+        // Create a dictionary to lookup the group names later. 
+        var groupNameDictionary = orgGroups.ToDictionary(x => x.Id, x => x.Name);
+
+        // Get collection counts into a dictionary
+        var groupCollectionItems = orgItems.SelectMany(x => x.CollectionIds, (x, b) => new { CipherId = x.Id, CollectionId = b }).GroupBy(y => y.CollectionId, (key, g) => new { CollectionId = key, ItemCount = g.Count() });
```

Comment:
This is called `groupCollectionItems` but it's not related to groups - I assume it refers to the use of `GroupBy`, but it's a confusing name. Maybe just `collectionItemCounts`?

I didn't know that you could use `SelectMany` in this way, which is cool. However this is a super long line, it would help comprehension to break it across multiple lines, e.g. (just because I already did it for review)

```cs
        var groupCollectionItems = orgItems
            .SelectMany(x => x.CollectionIds,
                (x, b) => new { CipherId = x.Id, CollectionId = b })
            .GroupBy(y => y.CollectionId,
                (key, g) => new { CollectionId = key, ItemCount = g.Count() });
```